mapsTask/js/mapController.js

```diff
@@ -0,0 +1,122 @@
+
+function initialize() {
+    initMap();
+    initAutocomplete();
+}
+
+var map;
+var geocoder;
+var marker;
+var icon;
+
+//Placed marker when map is pinned
+function placeMarker(location) {
+    marker = new google.maps.Marker({
+        position: location,
+        icon: icon,
+        map: map
+    });
+    map.setCenter(location);
+}
+
+// When there is a marker - clear marker 
+function clearMarker(marker) {
+    if (marker) {
+        marker.setMap(null)
+    }
+}
+
+function initMap() {
+    icon = {
+        url: "image/pin.png",
+        size: new google.maps.Size(150, 150),
+        origin: new google.maps.Point(0, 0),
+        anchor: new google.maps.Point(0, 0),
+        scaledSize: new google.maps.Size(45, 45)
+    };
+
+//Load the map
+    geocoder = new google.maps.Geocoder();
+    var defaultlatLng = new google.maps.LatLng(42.692151, 23.354187);
+    var myOptions = {
+        zoom: 8,
+        center: defaultlatLng,
+    }
+    map = new google.maps.Map(document.getElementById("map"), myOptions);
+
+//Add click event to map
+    google.maps.event.addListener(map, 'click', function (event) {
+
+// Removes the markers from the map
+        clearMarker(marker);
+
+// Get possition of marker when map is pin    
+        placeMarker(event.latLng);
+        var latlng = {
+            lat: event.latLng.lat(),
+            lng: event.latLng.lng()
+        }
+
+// Show address in the input
+        geocoder.geocode({ 'location': latlng }, function (results, status) {
+            if (status === 'OK') {
+                address = results[0].formatted_address
+                document.getElementById("address").value = address;
+            } else {
+                window.alert('Geocoder failed due to: ' + status);
```

Comment:
Every object that is attached to the window is a global one. Meaning you can call it directly without getting it from `window`. 

`window.alert('Hi')` is the same as `alert('Hi')`

Using alerts to notify users about status of your program is not very practical. Please find another, more appropriate way.